app/src/lib/git/submodule.ts

```diff
@@ -21,56 +21,42 @@ export async function listSubmodules(
   // We don't recurse when listing submodules here because we don't have a good
   // story about managing these currently. So for now we're only listing
   // changes to the top-level submodules to be consistent with `git status`
-  const result = await git(
+  const { stdout, exitCode } = await git(
     ['submodule', 'status', '--'],
     repository.path,
     'listSubmodules',
-    {
-      successExitCodes: new Set([0, 128]),
-    }
+    { successExitCodes: new Set([0, 128]) }
   )
 
-  if (result.exitCode === 128) {
+  if (exitCode === 128) {
     // unable to parse submodules in repository, giving up
     return []
   }
 
   const submodules = new Array<SubmoduleEntry>()
 
-  for (const entry of result.stdout.split('\n')) {
-    if (entry.length === 0) {
-      continue
-    }
+  // entries are of the format:
+  //  1eaabe34fc6f486367a176207420378f587d3b48 git (v2.16.0-rc0)
+  //
+  // first character:
+  //   - " " if no change
+  //   - "-" if the submodule is not initialized
+  //   - "+" if the currently checked out submodule commit does not match the SHA-1 found in the index of the containing repository
+  //   - "U" if the submodule has merge conflicts
+  //
+  // then the 40-character SHA represents the current commit
+  //
+  // then the path to the submodule
+  //
+  // then the output of `git describe` for the submodule in braces
+  // we're not leveraging this in the app, so go and read the docs
+  // about it if you want to learn more:
+  //
+  // https://git-scm.com/docs/git-describe
+  const statusRe = /^.([a-f0-9]{40}) (.+?) \((.+?)\)$/gm
```

Comment:
Not an actual problem, just a comment/thought: I usually try to avoid such specific regexp like the one for commit SHAs when it's not needed, in case the format changes somehow (shorter/longer shas now or in the future… whatever), and just look for non-whitespace characters, non-space characters in this case:
```suggestion
  const statusRe = /^.([^ ]+) (.+?) \((.+?)\)$/gm
```

Feel free to ignore this comment though (specially if my regexp is wrong 😂 )

Reply:
Also it seems it doesn't work for folders with `()` in them:

```javascript
Array.from(' 1eaabe34fc6f486367a176207420378f587d3b48 this is a long (folder name) (v2.16.0-rc0)'.matchAll(/^.([a-f0-9]{40}) (.+?) \((.+?)\)$/gm))
```
![image](https://user-images.githubusercontent.com/1083228/156169592-a5b6e1d4-b4cd-4d76-b2df-997e196ebd6e.png)

Maybe it's not even worth to worry about this, though, but I think just making the middle part greedy fixes it:
```suggestion
  const statusRe = /^.([^ ]+) (.+) \((.+?)\)$/gm
```

![image](https://user-images.githubusercontent.com/1083228/156169840-3f513e84-3383-489c-aa32-16f6c8c2676e.png)

What do you think?

Reply:
I think both your suggestions are A+! :sparkles: